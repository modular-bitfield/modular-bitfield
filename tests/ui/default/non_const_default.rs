use modular_bitfield::prelude::*;

fn get_value() -> u8 {
    42
}

#[bitfield]
pub struct NonConstDefault {
    #[default(get_value())]  // Function call not allowed in const context
    field: B8,
}

fn main() {}