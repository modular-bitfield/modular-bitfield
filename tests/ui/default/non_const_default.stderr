error: encountered invalid format for #[default] field attribute, expected #[default = expression]
 --> tests/ui/default/non_const_default.rs:9:7
  |
9 |     #[default(get_value())]  // Function call not allowed in const context
  |       ^^^^^^^^^^^^^^^^^^^^
